src/main/java/teammates/ui/controller/InstructorSearchPageData.java

```diff
@@ -58,4 +139,272 @@ public String getStudentRecordsLink(String courseId, StudentAttributes student){
+                              + frc.courseId + "#" + frc.getId();         
+            ElementTag editButton = createEditButton(link, Const.Tooltips.COMMENT_EDIT_IN_COMMENTS_PAGE);
+            
+            rows.add(new FeedbackResponseCommentRow(frCommentGiver, frc, creationTime, editButton));
+        } 
+        return rows;
+    }
+    
+    private List<StudentRow> createStudentRows(String courseId) {
+        List<StudentRow> rows = new ArrayList<StudentRow>();      
+        List<StudentAttributes> studentsInCourse = filterStudentsByCourse(courseId);
+        
+        for (StudentAttributes student : studentsInCourse) {
+            String viewPhotoLink = addUserIdToUrl(student.getPublicProfilePictureUrl());        
+            String actions = getStudentActions(student);
+
+            rows.add(new StudentRow(viewPhotoLink, student.section, student.team, student.name, student.email, actions));
+        }
+        return rows;
+    }
+    
+    /**
+     * Filters students from studentSearchResultBundle by course ID
+     * @param courseId 
+     * @return students whose course ID is equal to the courseId given in the parameter
+     */
+    private List<StudentAttributes> filterStudentsByCourse(String courseId) {
+        List<StudentAttributes> students = new ArrayList<StudentAttributes>();
+        
+        for (StudentAttributes student : studentSearchResultBundle.studentList) {
+            if (courseId.equals(student.course)) {
+                students.add(student);
+            }
+        }
+        return students;
+    }
+
+    private ElementTag createEditButton(String href, String tooltip) {
+        return new ElementTag("href", href, "title", tooltip);
+    }
+    
+    /**
+     * Returns HTML string for student action links - view, edit, delete, all records, add comment
+     * when an instructor searches for a student
+     * @param student Details of the student
+     * @return HTML output for the links
+     */
+    public String getStudentActions(StudentAttributes student) {
+        StringBuilder result = new StringBuilder();
+        InstructorAttributes instructor = studentSearchResultBundle.instructors.get(student.course);
+        String disabledStr = "";
+        
+        // View       
+        if (!instructor.isAllowedForPrivilege(student.section, Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_STUDENT_IN_SECTIONS)) {
+            disabledStr = "disabled=\"disabled\"";
+        } else {
+            disabledStr = "";
+        }
+        
+        result.append("<a class=\"btn btn-default btn-xs student-view-for-test\" "
+                       + "href=\"" + getCourseStudentDetailsLink(student.course, student) + "\" "
+                       + "title=\"" + Const.Tooltips.COURSE_STUDENT_DETAILS + "\" "
+                       + "data-toggle=\"tooltip\" data-placement=\"top\" "
+                       + disabledStr + ">");
+        result.append("View");
+        result.append("</a> ");
```

Comment:
@damithc Would it be okay for us to introduce HTML in Java? Or should we avoid it as much as possible?